### ISSUE TITLE
🚀 Feature: Allow document general attributes on update

### ISSUE THREAD

**Issue by silberGuy** [appwrite/appwrite#4802]
### 🔖 Feature description

When updating a document from my web app, sometimes we update the document object itself and pass it whole to the update function.
This procedure (which worked before updating from 0.14 to 1.1) now throws the following:
```
AppwriteException: Invalid document structure: Unknown attribute: "$collectionId"
    at Client.<anonymous> (http://localhost:3333/node_modules/.vite/deps/appwrite.js?v=61b2c4de:775:17)
    at Generator.next (<anonymous>)
    at fulfilled (http://localhost:3333/node_modules/.vite/deps/appwrite.js?v=61b2c4de:488:24)
```

I think that in case that the user of the update function is not trying to change this property, the backend can ignore it.

### 🎤 Pitch

I guess that in general, attributes like `$collectionId` should not be used, for the `$` prefix indicates a system-level property.
My current solution was to filter-out the property, and also `$permissions`, `$id` etc, on client-side level.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@silberGuy thanks for raising this issue! Could you please provide sample code for this?

**Comment by silberGuy** [comment]
Well, I am not much of a PHP dev,
my client side JS code looks similar to the following:

```js
async function updateDoc(collectionId: string, docId: string, docData: Partial<DocType>, permissions?: string[]) {
  const updateData = {
      ...docData,
      id: undefined,
      $collectionId: undefined,
      $createdAt: undefined,
      $databaseId: undefined,
      $id: undefined,
      $permissions: undefined,
  }
  return await appwriteDbs.updateDocument(
    'default', collectionId, docId, updateData, permissions,
  )
}
```

and I would love to just use:
```js
async function updateDoc(collectionId: string, docId: string, docData: Partial<DocType>, permissions?: string[]) {
  return await appwriteDbs.updateDocument(
    'default', collectionId, docData, updateData, permissions,
  )
}
```

I guess we can add similar filtering logic in the _JS sdk_.
But, I think that to solve this for all sdks (I did not use any other though), better solutions could be:
- The server would filter out these keys
- The server would throw only when the client attempts to change their values

**Comment by JakeAi** [comment]
@stnguyen90 I agree with @silberGuy. It used to be this way. Now I can send all of the "$" attributes except $databaseId and $collectionId... pick a theme and stick to it please. Having to remove all of the $attributes before updating a document in so many different areas is tiresome and repeating and too many areas to miss.

**Comment by eldadfux** [comment]
Stale issue, closing down.
